packages/lesswrong/server/callbacks/commentCallbacks.ts

```diff
@@ -586,17 +590,51 @@ getCollectionHooks("Comments").updateAsync.add(async function updatedCommentMayb
   await triggerReviewIfNeeded(currentUser._id)
 });
 
-getCollectionHooks("Comments").updateAsync.add(async function updateUserNotesOnCommentRejection ({ document, oldDocument, currentUser, context }: UpdateCallbackProperties<"Comments">) {
-  if (!oldDocument.rejected && document.rejected) {
-    void createMutator({
-      collection: context.ModeratorActions,
-      context,
-      currentUser,
-      document: {
-        userId: document.userId,
-        type: REJECTED_COMMENT,
-        endedAt: new Date()
-      }
+/**
+ * Run side effects based on the `forumEventMetadata` that is submitted.
+ */
+async function forumEventSideEffects({ comment, forumEventMetadata }: { comment: DbComment; forumEventMetadata: ForumEventCommentMetadata; }) {
+  if (forumEventMetadata.eventFormat === "STICKERS") {
+    const sticker = forumEventMetadata.sticker
+
+    if (!comment.forumEventId) {
+      throw new Error("Comment must have forumEventId")
+    }
+
+    const {_id, x, y, theta, emoji} = sticker ?? {};
+
+    if (!sticker || !_id || !x || !y || !theta) {
+      throw new Error("Must include sticker")
+    }
+
+    if (!emoji) {
+      throw new Error("No emoji selected")
+    }
+
+    const forumEventId = comment.forumEventId;
+    const stickerData = {_id, x, y, theta, emoji, commentId: comment._id, userId: comment.userId};
+
+    await new ForumEventsRepo().addSticker({ forumEventId, stickerData });
+    captureEvent("addForumEventSticker", {
+      forumEventId,
+      stickerData,
     });
   }
+}
+
+getCollectionHooks("Comments").newSync.add(async (comment: DbComment, _, context: ResolverContext) => {
+  if (comment.forumEventMetadata) {
+    // Side effects may need to reference the comment, so set the _id now
+    comment._id = comment._id || randomId();
+    await forumEventSideEffects({ comment, forumEventMetadata: comment.forumEventMetadata });
+  }
+  return comment;
+});
+
+getCollectionHooks("Comments").editSync.add(async (modifier, comment: DbComment) => {
+  const newMetadata = modifier.$set?.forumEventMetadata;
+  if (newMetadata && !isEqual(comment.forumEventMetadata, newMetadata)) {
+    await forumEventSideEffects({ comment, forumEventMetadata: newMetadata });
+  }
+  return modifier
```

Comment:
Why do we need this? I don't think any of the sticker data can be edited, right?

Reply:
Right, to clarify: I understand that's the reasoning for `newSync` above this, but I don't get when the `editSync` is used?